### PR TITLE
Automatically build document and publish to Github Pages

### LINKED ISSUE #179: Setup auto-release documents
# Problem Statement

We should auto-maintain the document release process and keep updating the docs.

- Should we use [mkdocs](https://www.mkdocs.org/) or [rst](https://docutils.sourceforge.io/rst.html)?
- Theme? 
- Host service (https://readthedocs.org/)?

### PR DESCRIPTION
**Pull request by eddyxu**
Closes #179

### CODE REVIEW

**Comment by eddyxu** [comment]
Rtd build out of this branch

https://lance.readthedocs.io/en/lei-docs/start/

**Comment by eddyxu** [comment]
Publish to https://eto-ai.github.io/lance/

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          cache-dependency-path: "docs/requirements.txt"
+      - name: Install dependencies
+        run: |
+          sudo apt install -y -qq doxygen
+      - name: Build Python
+        working-directory: python
+        run: |
+          ./tools/build_wheel.sh cp310
+          python -m pip install -r ../docs/requirements.txt
+          python -m pip install $(ls wheels/*.whl)
```
Comment (changhiskhan) [inline .github/workflows/docs.yml:45]: Should we install the wheels first so their versions take precedence?

**Review by changhiskhan — APPROVED** [review]

File: .github/workflows/docs.yml (hunk 1 above)
Comment (eddyxu) [reply .github/workflows/docs.yml:45]: Sure, can do

**Pull request merged**